Orange/data/table/base.py

```diff
@@ -0,0 +1,1292 @@
+        # which will then be processed by the if clause a couple of lines above this
+        if len(args) == 1 and (isinstance(args[0], (pd.core.internals.BlockManager, dict))
+                               or (isinstance(args[0], np.ndarray) and len(kwargs) != 0 and all_kwargs_are_pandas)):
+            return cls(data=args[0], **kwargs)
+        ##### END PANDAS SUBCLASS COMPATIBILITY SECTION #####
+
+        if 'filename' in kwargs:
+            args = [kwargs.pop('filename')]
+
+        if isinstance(args[0], str):
+            if args[0].startswith('https://') or args[0].startswith('http://'):
+                return cls.from_url(args[0], **kwargs)
+            else:
+                return cls.from_file(args[0], **kwargs)
+        elif isinstance(args[0], TableBase):
+            return cls.from_table(args[0].domain, args[0])
+        elif isinstance(args[0], pd.DataFrame):
+            return cls.from_dataframe(None, args[0], **kwargs)
+        elif isinstance(args[0], Domain):
+            domain, args = args[0], args[1:]
+            if not args:
+                return cls.from_domain(domain, **kwargs)
+            if isinstance(args[0], TableBase):
+                return cls.from_table(domain, *args)
+            elif isinstance(args[0], list):
+                return cls.from_list(domain, *args)
+            elif isinstance(args[0], pd.DataFrame):
+                return cls.from_dataframe(domain, args[0], **kwargs)
+        else:
+            domain = None
+        return cls.from_numpy(domain, *args, **kwargs)
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the object.
+
+        Sets the name and other attributes and calls pandas' initialization.
+        Filters the domain: retains only the columns actually in the table.
+        Sets the weights if they are not set already.
+        """
+        # see the comment in __new__ for the rationale here
+        # also, another tidbit is that pandas has some internals that need to be set up
+        # and expects its arguments to be set appropriately
+        # because we override the constructor arguments in a completely incompatible way,
+        # we need to pass ourselves as the data object if we have already set things up
+        # previously by e.g. creating an empty Table via the __new__ hack in from_X
+        # functions, then filling up with columns.
+        # to check for this, we check for domain existence because tables without domains
+        # can't really be used in Orange in any meaningful way
+        if hasattr(self, 'domain'):
+            kwargs['data'] = self
+
+        # only pass through things known to pandas, e.g.
+        # Table(..., weights=1) passes weights to from_numpy, but would error
+        # when passing to pandas upstream
+        super().__init__(**{k: v for k, v in kwargs.items() if k in self._KNOWN_PANDAS_KWARGS})
+
+        self.name = getattr(self, 'name', kwargs.get("name", "untitled"))
```

Comment:
Nope, a legacy remnant. But `getattr` must exist because the table may already have a name set, I just removed the kwargs part.